src/Engine/FloodFillFunctor.java

```diff
@@ -0,0 +1,69 @@
+package Engine;
+
+import Terrain.GameMap;
+import Units.Unit;
+import Units.MoveTypes.MoveType;
+
+/**
+ * Provides a stateless interface for determining traversability
+ * Used in {@link #findShortestPath(XYCoord, Unit, int, int, GameMap, boolean) findShortestPath()} and {@link #findPossibleDestinations(XYCoord, Unit, GameMap, boolean) findPossibleDestinations()}
+ */
+public interface FloodFillFunctor
+{
+  /**
+   * Determines whether the Location (x, y), can be traveled through.
+   * @param ignoreUnits If set, this function will ignore enemy-unit presence.
+   */
+  int getRemainingFillPower(GameMap map, int initialFillPower, XYCoord from, XYCoord to);
```

Comment:
Why `getRemainingFillPower`? That kind of makes it sounds like FFF is managing the fill power internally. It seems more natural to use something like `getTransitionCost`, and then we don't even need to pass in a starting fill power.
`newPower = getRemainingFillPower(oldPower)`
would become
`newPower = oldPower -= getTransitionCost()`
which frankly reads more naturally to me.